sound/soc/sof/virtio-fe.c

```diff
@@ -0,0 +1,835 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2017-2019 Intel Corporation. All rights reserved.
+ *
+ * Author:	Libin Yang <libin.yang@intel.com>
+ *		Luo Xionghu <xionghu.luo@intel.com>
+ *		Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ *		Guennadi Liakhovetski <guennadi.liakhovetski@linux.intel.com>
+ */
+
+/*
+ * virt IO FE driver
```

Comment:
@lyakh sorry I might be bothering you with a lot of "VirtIO for Dummies" questions. But what is a Virtio FE and BE really?

Reply:
@ranj063 Front-end and back-end. Front end is the guest, back end is the host. The front end is always a driver, ATAIK, but the back end can be one of the 3 types: a kernel driver, a user-space driver, or pass-through.

Reply:
@lyakh thanks. Does the BE type depend on the implementation? What type is it in the case of SOF?

Reply:
@ranj063 Yes, all 3 are different, you decide what you implement, in our case it's of course the in-kernel a.k.a. the vhost type.